code/game/machinery/turret_control.dm

```diff
@@ -180,9 +180,32 @@
 	TC.ailock = ailock
 
 	if(istype(control_area))
+		var/lethal_only = 0 // To check if there is a lethal turret in the area.
+		var/both_mode = 0
 		for (var/obj/machinery/porta_turret/aTurret in control_area)
+			// If turret only has lethal mode - lock switching modes
+			if(aTurret.egun == 0)
+				lethal_only = 1
+				egun = 0
+			else
+				both_mode = 1
 			aTurret.setState(TC)
-
+		// If there is a turret with lethal mode only, and turret with both modes. Scan for lethal turrets, and disable them.
+		if(both_mode && lethal_only)
+			if(lethal)
+				for (var/obj/machinery/porta_turret/bTurret in control_area)
+					bTurret.setState(TC)
```

Comment:
You're setting state for the second time. Again, more reasons to use this proc purely for setting the state (and thus not modifying egun or lethal) and another proc for updating available states.

Reply:
@skull132 Note it only sets state once, because this is the only loop that calls `setState`. So there is no redundancy, it just on-top of setting state will make lethal turrets disabled if we have mix of them, and are in stun mode.

Reply:
State is also set in the loop on line 192. Probably because you forgot to remove that call.

Reply:
I did